core/primitives/src/action/mod.rs

```diff
@@ -178,15 +178,20 @@ pub enum Action {
     DeleteAccount(DeleteAccountAction),
     Delegate(Box<delegate::SignedDelegateAction>),
 }
-// Note: If this number ever goes down, please adjust the equality accordingly. Otherwise,
-// we would get used to better performance and would be subject to a performance loss should
-// we come back up to 32 bytes later on.
-// If compiling with nightly, this check may fail due to new optimizations introduced by
-// rustc. So, cfg-them out, as our production system only runs with stable.
-#[cfg(not(any(fuzz, udeps)))]
+
+// Note: If this number ever goes down, please adjust the comparison accordingly. Otherwise,
+// we might get used to better performance and would be subject to a performance loss should we
+// come back up to 32 bytes later on.
 const _: () = assert!(
-    cfg!(not(target_pointer_width = "64")) || std::mem::size_of::<Action>() == 32,
-    "Action is 32 bytes for performance reasons, see #9451"
+    std::mem::size_of::<Action>() <= (if cfg!(any(fuzz, udeps)) {
+        // Nightly now did result in a size improvement. This check right here is checked by our CI
+        // to make sure no regression from 24 bytes back to 32 bytes on nightly occurs without our
+        // knowledge.
+        24
```

Comment:
The one downside is that the size could further decrease (to e.g. 16 bytes) and we would be none-the-wiser, but I find that to be quite unlikely.